Tabloid/client/src/components/PostEdit.js

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+import { useParams } from "react-router-dom";
+import { Container, Form, FormGroup, Label, Input, Button } from "reactstrap";
+import { getAllCategories } from "../modules/categoryManager.js";
+import { getPostById, editPost } from "../modules/postManager.js";
+
+export const PostEdit = () => {
+    const [post, setPost] = useState({});
+    const [categories, setCategories] = useState([]);
+    const { id } = useParams();
+    const history = useHistory();
+
+    const handleInputChange = (evt) => {
+        const value = evt.target.value;
+        const key = evt.target.id;
+
+        const postCopy = { ...post };
+
+        postCopy[key] = value;
+        setPost(postCopy);
+    };
+
+    const updatePost = () => {
+        editPost(post).then(() => history.push(`/posts/${post.id}`));
+    };
+
+    useEffect(() => {
+        getPostById(id).then(setPost);
+        getAllCategories().then(setCategories);
+    }, []);
```

Comment:
This dependency array is empty right now, but it should probably watch for a change to the id variable.